index.js

```diff
@@ -0,0 +1,254 @@
+/**
+ * This plugin allows us to create shared chunks which contain all module
+ * dependencies that don't already exist in an existing shared chunk.
+ *
+ * All dependencies required to load a shared chunk appear in the 'parents'
+ * array on each chunk.
+ *
+ * A chunk is webpack terminology for a set of modules (.js files).  Entry
+ * chunks are special chunks which initially contain the entry point (a .js
```

Comment:
Can the entry point .js file ever get moved out of its chunk and in to another? Or is it guaranteed to always be standalone?

Reply:
I supposed you could have a couple of entry points which have a third entry point as a dependency and depending on the settings it might get pulled out into a shared chunk.  This would be bad because currently we don't emit empty chunks.  I'll write a test case for that.  I think for now the easiest thing to do would be to allow empty entry chunks.